shared/src/commonTest/kotlin/com/mbta/tid/mbta_app/viewModel/MapViewModelTests.kt

```diff
@@ -139,4 +169,154 @@ class MapViewModelTests : KoinTest {
+        val layerManger = mock<IMapLayerManager>(MockMode.autofill)
+
+        val objects = TestData.clone()
+        val chestnutHill = objects.getStop("place-chill")
+        val southStreet = objects.getStop("place-sougr")
+        val informedEntities =
+            listOf(chestnutHill, southStreet).flatMap {
+                listOf(
+                    Alert.InformedEntity(
+                        listOf(
+                            Alert.InformedEntity.Activity.Board,
+                            Alert.InformedEntity.Activity.Ride,
+                        ),
+                        stop = it.id,
+                        route = "Green-B",
+                    )
+                )
+            }
+        val alert =
+            objects.alert {
+                effect = Alert.Effect.Suspension
+                activePeriod =
+                    mutableListOf(Alert.ActivePeriod(Clock.System.now().minus(5.hours), null))
+                informedEntity = informedEntities.toMutableList()
+            }
+
+        testViewModelFlow(viewModel).test {
+            viewModel.layerManagerInitialized(layerManger)
+            viewModel.setViewportManager(viewportProvider)
+            viewModel.densityChanged(1f)
+            assertEquals(MapViewModel.State.Overview, awaitItem())
+            advanceUntilIdle()
+            viewModel.alertsChanged(AlertsStreamDataResponse(objects))
+            advanceUntilIdle()
+        }
+
+        advanceUntilIdle()
+        verifySuspend(VerifyMode.order) {
+            // 3 times for overview  - 2 for initial setup, 1 for alerts changing
+            layerManger.updateRouteSourceData(matching { it.size == 6 })
+            layerManger.updateRouteSourceData(matching { it.size == 6 })
+            layerManger.updateRouteSourceData(matching { it.size == 6 })
+        }
+    }
+
+    @Test
+    @Ignore // TODO address flakiness
+    fun allRailLayersResetWhenNavigatingToNearby() = runTest {
+        val dispatcher = StandardTestDispatcher(testScheduler)
+        setUpKoin(dispatcher)
+
+        val viewportProvider = MockViewportManager()
+        val viewModel: MapViewModel = get()
+        val layerManger = mock<IMapLayerManager>(MockMode.autofill)
+
+        val objects = TestData.clone()
+        val chestnutHill = objects.getStop("place-chill")
+
+        testViewModelFlow(viewModel).test {
+            viewModel.setViewportManager(viewportProvider)
+            viewModel.layerManagerInitialized(layerManger)
+            viewModel.densityChanged(1f)
+            assertEquals(MapViewModel.State.Overview, awaitItem())
+            advanceUntilIdle()
+            viewModel.navChanged(SheetRoutes.StopDetails(chestnutHill.id, null, null))
+            assertEquals(MapViewModel.State.StopSelected(chestnutHill, null), awaitItem())
+            advanceUntilIdle()
+        }
+
+        advanceUntilIdle()
+        viewModel.navChanged(SheetRoutes.NearbyTransit)
+        advanceUntilIdle()
```

Comment:
This test passes when run on its own if this `viewModel.navChange` call is moved outside the `testViewModelFlow`, but still fails when run in a batch. Not clear to me why calling this within `testViewModelFlow` would *never* pass - there maybe some fundamental of molecule or async testing that I'm missing here.